[CI] ESQL Centroid tests using no doc values failing on 8.x

In https://github.com/elastic/elasticsearch/issues/116373 the build scans show a disconnecting node due to an AssertionError:
```
[2024-11-18T13:44:38,686][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-2] fatal error in thread [elasticsearch[test-cluster-2][esql_worker][T#1]], exiting java.lang.AssertionError: BYTES_REF NOT IN (NULL, LONG)
```

This happens in some tests added in https://github.com/elastic/elasticsearch/pull/116800:
- spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNotIndexedNorDocValues
- spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNoDocValues
I.e. only the "no doc values" cases added in the PR.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -179,24 +179,6 @@ tests:
   - class: org.elasticsearch.threadpool.SimpleThreadPoolIT
     method: testThreadPoolMetrics
     issue: https://github.com/elastic/elasticsearch/issues/108320
-  - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNoDocValues SYNC}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
-  - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNoDocValues ASYNC}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
-  - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNoDocValues}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
-  - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNotIndexedNorDocValues SYNC}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
-  - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNotIndexedNorDocValues ASYNC}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
-  - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-    method: test {spatial.CentroidFromAirportsAfterIntersectsCompoundPredicateNotIndexedNorDocValues}
-    issue: https://github.com/elastic/elasticsearch/issues/116945
   - class: org.elasticsearch.xpack.inference.InferenceRestIT
     method: test {p0=inference/40_semantic_text_query/Query a field that uses the default ELSER 2 endpoint}
     issue: https://github.com/elastic/elasticsearch/issues/117027

